### PR TITLE
Add RDS Public Virtual Interface query for AWS CloudFormation

### LINKED ISSUE #1168: Add RDS Public Virtual Interface query for AWS CloudFormation
### Platform
AWS CloudFormation

### Provider
AWS

### Description
Amazon RDS databases wit direct connect virtual interfaces must not have public interfaces, which means the IP Address in its Security Group must not match prefixes advertised by Amazon.

### PR DESCRIPTION
**Pull request by Ruben-Silva**
Adding RDS Public Virtual Interface query for AWS CloudFormation, that checks if the IP Address in its Security Group matchs prefixes advertised by Amazon.

Closes #1168

### CODE REVIEW

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      - IpProtocol: tcp
+        FromPort: 80
+        ToPort: 80
+        CidrIp: 0.0.0.0/0
+  DBInstance3:
+    Type: AWS::RDS::DBInstance
+    Properties:
+      PubliclyAccessible: true
+      DBName:
+        Ref: DBName
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml:79]: `E1012 Ref DBName not found as a resource or parameter`

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      Engine: MySQL
+      MultiAZ:
+        Ref: MultiAZDatabase
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml:82]: `E1012 Ref MultiAZDatabase not found as a resource or parameter`

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      MasterUsername:
+        Ref: DBUser
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml:84]: `E1012 Ref DBUser not found as a resource or parameter`

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      DBInstanceClass:
+        Ref: DBClass
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml:86]: `E1012 Ref DBClass not found as a resource or parameter`

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      AllocatedStorage:
+        Ref: DBAllocatedStorage
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml:88]: `E1012 Ref DBAllocatedStorage not found as a resource or parameter`

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      MasterUserPassword:
+        Ref: DBPassword
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml:90]: `E1012 Ref DBPassword not found as a resource or parameter`

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  DBEC2SecurityGroup2:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Open database for access
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 80
+        ToPort: 80
+        CidrIpv6: 2600:1f70:4000::/56
+      SecurityGroupEgress:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml:69]: `E2521 Property VpcId should exist with SecurityGroupEgress for Resources/DBEC2SecurityGroup2/Properties`

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  DBEC2SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Open database for access
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 80
+        ToPort: 80
+        CidrIp: 3.5.140.0/22
+      SecurityGroupEgress:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml:12]: `E2521 Property VpcId should exist with SecurityGroupEgress for Resources/DBEC2SecurityGroup/Properties`

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/negative.yaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  DBEC2SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Open database for access
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 80
+        ToPort: 80
+        CidrIp: 1.2.3.4/24
+      SecurityGroupEgress:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/negative.yaml:12]: `E2521 Property VpcId should exist with SecurityGroupEgress for Resources/DBEC2SecurityGroup/Properties`

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+      - IpProtocol: tcp
+        FromPort: 80
+        ToPort: 80
+        CidrIp: 0.0.0.0/0
+  DBInstance:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml:17]: `W3011 Both UpdateReplacePolicy and DeletionPolicy are needed to protect Resources/DBinstance from deletion`

```
DeletionPolicy: Retain
UpdateReplacePolicy: Retain
```

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/negative.yaml [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+      - IpProtocol: tcp
+        FromPort: 80
+        ToPort: 80
+        CidrIp: 1.2.3.4/24
+  DBInstance:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/negative.yaml:17]: `W3011 Both UpdateReplacePolicy and DeletionPolicy are needed to protect Resources/DBinstance from deletion`

```
DeletionPolicy: Retain
UpdateReplacePolicy: Retain
```

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+        Ref: DBClass
+      AllocatedStorage:
+        Ref: DBAllocatedStorage
+      MasterUserPassword:
+        Ref: DBPassword
+      VPCSecurityGroups:
+      - !GetAtt DBEC2SecurityGroup.GroupId
+---
+Resources:
+  DBinstance2:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml:38]: `W3011 Both UpdateReplacePolicy and DeletionPolicy are needed to protect Resources/DBinstance2 from deletion`

```
UpdateReplacePolicy: Retain
```

File: assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+      - IpProtocol: tcp
+        FromPort: 80
+        ToPort: 80
+        CidrIpv6: 2600:1f70:4000::/56
+      SecurityGroupEgress:
+      - IpProtocol: tcp
+        FromPort: 80
+        ToPort: 80
+        CidrIp: 0.0.0.0/0
+  DBInstance3:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_public_virtual_interface/test/positive.yaml:74]: `W3011 Both UpdateReplacePolicy and DeletionPolicy are needed to protect Resources/DBinstance3 from deletion`

```
DeletionPolicy: Retain
UpdateReplacePolicy: Retain
```

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
Requested changes

**Comment by Ruben-Silva** [comment]
This query can't be done through static analysis so its being removed.

**Pull request closed without merging**